NEWS.md

```diff
@@ -6,9 +6,27 @@
 
 ## NEW FEATURES
 
-1. New function `fcase(...,default)` implemented in C by Morgan Jacob, [#3823](https://github.com/Rdatatable/data.table/issues/3823), is inspired by SQL `CASE WHEN` which is a common tool in SQL for e.g. building labels or cutting age groups based on conditions. `fcase` is comparable to R function `dplyr::case_when`. Please see `?data.table::fcase` for more details.
+1. New function `fcase(...,default)` implemented in C by Morgan Jacob, [#3823](https://github.com/Rdatatable/data.table/issues/3823), is inspired by SQL `CASE WHEN` which is a common tool in SQL for e.g. building labels or cutting age groups based on conditions. `fcase` is comparable to R function `dplyr::case_when` however it evaluates its arguments in a lazy way (i.e. only when needed) as shown below. Please see `?data.table::fcase` for more details.
```

Comment:
> Please see `?data.table::fcase` for more details.

can be

> Please see `?fcase` for more details.

no need for namespace prefix here, it make sense when we compare multiple packages, but here we just point to documentation